runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillStateCache.java

```diff
@@ -225,24 +225,23 @@ public long getWeight() {
 
   /** Entry in the state cache that stores a map of values. */
   private static class StateCacheEntry implements Weighted {
-    private final HashMap<NamespacedTag<?>, WeightedValue<?>> values;
+    private final IdentityHashMap<InternedByteString, WeightedValue<?>> values;
```

Comment:
It seems like this might have collisions between types. For example if we have a StateTag<Value> StateTag<Bag> for the same "tag", I believe the bytestring for both would be the same and we could collide in the cache.

Can you add a test covering that?